lib/matplotlib/pyplot.py

```diff
@@ -1125,7 +1201,7 @@ def cla():
 ## More ways of creating axes ##
 
 @_docstring.dedent_interpd
-def subplot(*args, **kwargs):
+def subplot(*args, **kwargs) -> matplotlib.axes.Axes:
```

Comment:
This was admittedly an attempt to get sphinx to be quieter before I just ignored the warnings (`Axes` has multiple targets, and sphinx doesn't resolve which one is actually used, it just looks at the string)

It can (And probably should) be cleaned up and just ignored in sphinx

Reply:
Well, looks like my ignores didn't work anyway, those only work for missing references, not ambiguous ones.

So I added it back as `axes.Axes` to quiet sphinx warnings. Not as pretty, but better unwilling to turn of warnings -> errors in sphinx, so perhaps the best I can do at this time.

For reasons unclear to me, the some references do _not_ generate this warning, such as `sca`...

Reply:
well `axes.Axes` is no good because `axes` is defined in pyplot, so using the full `matplotlib.axes.Axes` instead, which is even more ugly, but best I've got